stable_baselines3/common/vec_env/__init__.py

```diff
@@ -35,36 +32,49 @@ def unwrap_vec_wrapper(env: Union["GymEnv", VecEnv], vec_wrapper_class: Type[Vec
     return None
 
 
-def unwrap_vec_normalize(env: Union["GymEnv", VecEnv]) -> Optional[VecNormalize]:
+def unwrap_vec_normalize(env: VecEnv) -> Optional[VecNormalize]:
     """
-    :param env:
-    :return:
+    Retrieve a ``VecNormalize`` object by recursively searching.
+
+    :param env: The VecEnv that is going to be unwrapped
+    :return: The ``VecNormalize`` object if the ``VecEnv`` is wrapped with ``VecNormalize``, None otherwise
     """
-    return unwrap_vec_wrapper(env, VecNormalize)  # pytype:disable=bad-return-type
+    return unwrap_vec_wrapper(env, VecNormalize)
 
 
-def is_vecenv_wrapped(env: Union["GymEnv", VecEnv], vec_wrapper_class: Type[VecEnvWrapper]) -> bool:
+def is_vecenv_wrapped(env: VecEnv, vec_wrapper_class: Type[VecEnvWrapper]) -> bool:
     """
     Check if an environment is already wrapped by a given ``VecEnvWrapper``.
 
-    :param env:
-    :param vec_wrapper_class:
-    :return:
+    :param env: The VecEnv that is going to be checked
+    :param vec_wrapper_class: The desired ``VecEnvWrapper`` class.
+    :return: True if the ``VecEnv`` is wrapped with the desired wrapper, False otherwise
     """
     return unwrap_vec_wrapper(env, vec_wrapper_class) is not None
 
 
-# Define here to avoid circular import
-def sync_envs_normalization(env: "GymEnv", eval_env: "GymEnv") -> None:
+def sync_envs_normalization(env: VecEnv, eval_env: VecEnv) -> None:
     """
-    Sync eval env and train env when using VecNormalize
+    Synchronize eval env and train env statistics when using ``VecNormalize``
 
-    :param env:
-    :param eval_env:
+    :param env: Training env
+    :param eval_env: Environment used for evaluation.
     """
     env_tmp, eval_env_tmp = env, eval_env
     while isinstance(env_tmp, VecEnvWrapper):
```

Comment:
Why don't we use the nice `unwrap_vec_env_wrapper` from above here?

Reply:
good point, I think this was to avoid doing two for loops and maybe raise an error when the two envs doesn't match.
But looking at it, I think you are right and we should switch to `unwrap_vec_env_wrapper`.

Reply:
let me check if I can find the issues where it was discussed

Reply:
See https://github.com/DLR-RM/stable-baselines3/issues/372 and linked issues like https://github.com/DLR-RM/stable-baselines3/issues/443#issuecomment-845784505

Reply:
what do you think?

Reply:
(you can reply here, but I think it should be part of another PR)